src/snakemake/dag.py

```
@@ -416,59 +417,100 @@ def update_conda_envs(self):
+                                and AccessPattern.RANDOM not in file_access_patterns
+                                and AccessPattern.MULTI not in file_access_patterns
+                                and None not in file_access_patterns
+                            )
+                            logger.debug(
+                                f"Ondemand eligibility of {f.storage_object.print_query}: "
+                                f"{f.storage_object.is_ondemand_eligible} with access "
+                                f"patterns {','.join(map(str, file_access_patterns))}"
+                            )
+
+                            async def retrieve_and_yield():
+                                await f.retrieve_from_storage()
```

Ruff (0.15.21)

[warning] Function definition does not bind loop variable `f`

(B023)